### ISSUE TITLE
License removed from setup.py is preventing proper metadata population

### ISSUE THREAD

**Issue by bellyhold** [cloud-custodian/cloud-custodian#6625]
**Describe the bug**
The license field in pypi is not being populated due to removal from setup.py. This is preventing my firm from acquiring the latest version for use due to our code scan tooling.

**Additional context**
This is a simple non-breaking change, but would help us immensely. 

** Proposed Fix **
```
setup_kwargs = {
    'name': 'c7n',
    'version': '0.9.12',
    'description': 'Cloud Custodian - Policy Rules Engine',
    'license': 'Apache-2.0',
    'long_description': 'removed for brevity...',
    'long_description_content_type': 'text/markdown',
    'author': 'Cloud Custodian Project',
    'author_email': None,
    'maintainer': None,
    'maintainer_email': None,
    'url': 'https://cloudcustodian.io',
    'packages': packages,
    'package_data': package_data,
    'install_requires': install_requires,
    'entry_points': entry_points,
    'python_requires': '>=3.6,<4.0',
}
```

**Comment by kapilt** [comment]
fix your code scanning tool  or whitelist :-) the license is apache 2.0 per the repo .. we're generating setup.py from poetry which doesn't include that metadata when publishing.

**Comment by bellyhold** [comment]
Ok. I'm realizing that setup.py is not the issue... in the wheel downloaded from pypi... the METADATA file is showing license as UNKNOWN. I understand this might not be an issue for everyone, but would you be willing to look into why the license isn't being populated correctly?

**Comment by kapilt** [comment]
so.. 
we have a license in pyproject.toml, 
we generate a frozen setup.py with tools/dev/poetrypkg.py (rel automation is out of the makefile)

tldr, its a code fix / debug on poetry re why we can't get license metadata into setup.py and wheel (wheel is via twine).

**Comment by kapilt** [comment]
pr up which injects license into the metadata when we monkeypatch poetry
